docs/diagrams/AddCommandOrAndXor.puml

```diff
@@ -0,0 +1,37 @@
+@startuml
```

Comment:
Needs revision - consider changing this to `BooleanTemplate` being associated with `Gate` objects, which all the other gates inherit from. Also, `TwoInputGate` only has `input` and `secondInput`, they do not know about other `Gate` objects. The logic is done from within `BooleanTemplate`.